Feature/tests private notes

Rudeka opened the pull request:
Added Tests for CreateNote, GetNote methods
Added Tests for Validator and Routes
Changed NoteQueries to work with IOrderedQueryable
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/96%23discussion_r61047401%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/96%23discussion_r61047574%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/96%23discussion_r61048153%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/96%23discussion_r61112582%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/96%23discussion_r61112729%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/96%23discussion_r61113769%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/96%23discussion_r61212156%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/96%23discussion_r61212309%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/96%23discussion_r61214285%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/96%23discussion_r61214288%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/96%23discussion_r61382395%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%207bd825394ab24e7182eefe955df7a45d2a730602%20DevelopGitHub/GitHubExtension/GitHubExtension.Notes.Tests/TestsForRoutes/NoteRouteTests.cs%2021%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/96%23discussion_r61048153%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Can%20you%20prevent%20code%20duplication%3F%22%2C%20%22created_at%22%3A%20%222016-04-26T08%3A41%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20managed%20to%20work%20correctly%20with%20routes%20from%20attributes%20only%20by%20calling%20GetRoutes%20method%20each%20time%20when%20url%20for%20a%20specific%20route%20is%20set%20and%20now%20I%20have%20to%20call%20it%20for%20each%20test.%20%20%20%22%2C%20%22created_at%22%3A%20%222016-04-26T16%3A03%3A04Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/14029898%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Rudeka%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Notes.Tests/TestsForRoutes/NoteRouteTests.cs%3AL1-34%22%7D%2C%20%22Pull%207bd825394ab24e7182eefe955df7a45d2a730602%20DevelopGitHub/GitHubExtension/GitHubExtension.Notes.Tests/Extensions/IdentityExtensions.cs%2012%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/96%23discussion_r61047401%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22You%20really%20need%20substitute%20for%20GenericIdentity%3F%22%2C%20%22created_at%22%3A%20%222016-04-26T08%3A36%3A37Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%2C%20%7B%22body%22%3A%20%22Changed%20to%20new%20instance%20instead%20of%20mock%20and%20commited%22%2C%20%22created_at%22%3A%20%222016-04-26T15%3A56%3A40Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/14029898%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Rudeka%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Notes.Tests/Extensions/IdentityExtensions.cs%3AL1-19%22%7D%2C%20%22Pull%20ad959c8a6c32e47f136eaae86602a08545612547%20DevelopGitHub/GitHubExtension/GitHubExtension.Notes.Tests/Extensions/IdentityExtensions.cs%209%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/96%23discussion_r61212156%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Please%20prevent%20code%20duplication%20code%20after%20rebase.%22%2C%20%22created_at%22%3A%20%222016-04-27T06%3A58%3A57Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20have%20renamed%20the%20class%20to%20TestsIdentityExtensions%20and%20commited%22%2C%20%22created_at%22%3A%20%222016-04-27T07%3A23%3A11Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/14029898%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Rudeka%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Notes.Tests/Extensions/IdentityExtensions.cs%3AL1-18%22%7D%2C%20%22Pull%207bd825394ab24e7182eefe955df7a45d2a730602%20DevelopGitHub/GitHubExtension/GitHubExtension.Notes.Tests/TestsForControllers/NoteControllerCreateNoteTests.cs%2024%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/96%23discussion_r61047574%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Data%20for%20what%20model%3F%22%2C%20%22created_at%22%3A%20%222016-04-26T08%3A37%3A33Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%2C%20%7B%22body%22%3A%20%22Changed%20to%20a%20more%20explicit%20name%20and%20commited%22%2C%20%22created_at%22%3A%20%222016-04-26T15%3A57%3A27Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/14029898%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Rudeka%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Notes.Tests/TestsForControllers/NoteControllerCreateNoteTests.cs%3AL1-106%22%7D%2C%20%22Pull%20ad959c8a6c32e47f136eaae86602a08545612547%20DevelopGitHub/GitHubExtension/GitHubExtension.Notes.Tests/TestsForValidator/NoteValidatorTests.cs%2010%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/96%23discussion_r61212309%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Where%20the%20positive%20test%3F%22%2C%20%22created_at%22%3A%20%222016-04-27T07%3A00%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%2C%20%7B%22body%22%3A%20%22There%20are%20positive%20tests%20below%20e.g.%20ShouldNotHaveErrorWhenBodyIsSpecified.%20I%20think%20it%20is%20better%20to%20rename%20it%20something%20like%20ValidModelWhenBodyIsSpecified.%20Made%20changes%20and%20commited.%22%2C%20%22created_at%22%3A%20%222016-04-27T07%3A23%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/14029898%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Rudeka%22%7D%7D%2C%20%7B%22body%22%3A%20%22Good%20idea%21%22%2C%20%22created_at%22%3A%20%222016-04-28T07%3A16%3A28Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Notes.Tests/TestsForValidator/NoteValidatorTests.cs%3AL1-38%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'><img src='http://www.codereviewhub.com/site/github-bar.png' height=40></a>
- [ ] <a href='#crh-comment-Pull 7bd825394ab24e7182eefe955df7a45d2a730602 DevelopGitHub/GitHubExtension/GitHubExtension.Notes.Tests/Extensions/IdentityExtensions.cs 12'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/96#discussion_r61047401'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Notes.Tests/Extensions/IdentityExtensions.cs:L1-19</a></b>
- <a href='https://github.com/TerraVenil'><img border=0 src='https://avatars.githubusercontent.com/u/6068068?v=3' height=16 width=16'></a> You really need substitute for GenericIdentity?
- <a href='https://github.com/Rudeka'><img border=0 src='https://avatars.githubusercontent.com/u/14029898?v=3' height=16 width=16'></a> Changed to new instance instead of mock and commited
- [ ] <a href='#crh-comment-Pull 7bd825394ab24e7182eefe955df7a45d2a730602 DevelopGitHub/GitHubExtension/GitHubExtension.Notes.Tests/TestsForControllers/NoteControllerCreateNoteTests.cs 24'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/96#discussion_r61047574'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Notes.Tests/TestsForControllers/NoteControllerCreateNoteTests.cs:L1-106</a></b>
- <a href='https://github.com/TerraVenil'><img border=0 src='https://avatars.githubusercontent.com/u/6068068?v=3' height=16 width=16'></a> Data for what model?
- <a href='https://github.com/Rudeka'><img border=0 src='https://avatars.githubusercontent.com/u/14029898?v=3' height=16 width=16'></a> Changed to a more explicit name and commited
- [ ] <a href='#crh-comment-Pull 7bd825394ab24e7182eefe955df7a45d2a730602 DevelopGitHub/GitHubExtension/GitHubExtension.Notes.Tests/TestsForRoutes/NoteRouteTests.cs 21'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/96#discussion_r61048153'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Notes.Tests/TestsForRoutes/NoteRouteTests.cs:L1-34</a></b>
- <a href='https://github.com/TerraVenil'><img border=0 src='https://avatars.githubusercontent.com/u/6068068?v=3' height=16 width=16'></a> Can you prevent code duplication?
- <a href='https://github.com/Rudeka'><img border=0 src='https://avatars.githubusercontent.com/u/14029898?v=3' height=16 width=16'></a> I managed to work correctly with routes from attributes only by calling GetRoutes method each time when url for a specific route is set and now I have to call it for each test.
- [ ] <a href='#crh-comment-Pull ad959c8a6c32e47f136eaae86602a08545612547 DevelopGitHub/GitHubExtension/GitHubExtension.Notes.Tests/Extensions/IdentityExtensions.cs 9'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/96#discussion_r61212156'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Notes.Tests/Extensions/IdentityExtensions.cs:L1-18</a></b>
- <a href='https://github.com/TerraVenil'><img border=0 src='https://avatars.githubusercontent.com/u/6068068?v=3' height=16 width=16'></a> Please prevent code duplication code after rebase.
- <a href='https://github.com/Rudeka'><img border=0 src='https://avatars.githubusercontent.com/u/14029898?v=3' height=16 width=16'></a> I have renamed the class to TestsIdentityExtensions and commited
- [ ] <a href='#crh-comment-Pull ad959c8a6c32e47f136eaae86602a08545612547 DevelopGitHub/GitHubExtension/GitHubExtension.Notes.Tests/TestsForValidator/NoteValidatorTests.cs 10'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/96#discussion_r61212309'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Notes.Tests/TestsForValidator/NoteValidatorTests.cs:L1-38</a></b>
- <a href='https://github.com/TerraVenil'><img border=0 src='https://avatars.githubusercontent.com/u/6068068?v=3' height=16 width=16'></a> Where the positive test?
- <a href='https://github.com/Rudeka'><img border=0 src='https://avatars.githubusercontent.com/u/14029898?v=3' height=16 width=16'></a> There are positive tests below e.g. ShouldNotHaveErrorWhenBodyIsSpecified. I think it is better to rename it something like ValidModelWhenBodyIsSpecified. Made changes and commited.
- <a href='https://github.com/TerraVenil'><img border=0 src='https://avatars.githubusercontent.com/u/6068068?v=3' height=16 width=16'></a> Good idea!

<a href='https://www.codereviewhub.com/ITsvetkoFF/Kv-013/pull/96?mark_as_completed=1'><img src='http://www.codereviewhub.com/site/github-mark-as-completed.png' height=26></a>&nbsp;<a href='https://www.codereviewhub.com/ITsvetkoFF/Kv-013/pull/96?approve=1'><img src='http://www.codereviewhub.com/site/github-approve.png' height=26></a>&nbsp;<a href='https://github.com/ITsvetkoFF/Kv-013/pull/96'><img src='http://www.codereviewhub.com/site/github-refresh.png' height=26></a>
<a href='#crh-end'></a>

Rudeka committed:
Add test project for notes

Rudeka committed:
Change NoteQueries to work with IOrderedQueryable

Rudeka committed:
Add Note Tests project

Rudeka committed:
Add tests for CreateNote method

Rudeka committed:
Add tests for GetNote method

Rudeka committed:
Add tests for validator

Rudeka committed:
Add tests for note routes

ITsvetkoFF commented:
This pull request is now managed with CodeReviewHub. <a href='https://github.com/ITsvetkoFF/Kv-013/pull/96'>Refresh</a>.<a href='#crh-update-none'></a>

TerraVenil commented on DevelopGitHub/GitHubExtension/GitHubExtension.Notes.Tests/Extensions/IdentityExtensions.cs:
```diff
@@ -0,0 +1,18 @@
+    {
+        public static void SetUserForController(this ApiController controller, string name, string typeOfClaim, string valueOfClaim)
+        {
+            var identity = Substitute.ForPartsOf<GenericIdentity>(name);
```
You really need substitute for GenericIdentity?

Rudeka replied:
Changed to new instance instead of mock and commited

TerraVenil commented on DevelopGitHub/GitHubExtension/GitHubExtension.Notes.Tests/TestsForControllers/NoteControllerCreateNoteTests.cs:
```diff
@@ -0,0 +1,105 @@
+        private const string TestCollaboratorId = "550e8400-e29b-41d4-a716-446655440000";
+        private const string TypeOfClaim = "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier";
+        
+        public static IEnumerable<object[]> DataForModel
```
Data for what model?

Rudeka replied:
Changed to a more explicit name and commited

TerraVenil commented on DevelopGitHub/GitHubExtension/GitHubExtension.Notes.Tests/TestsForRoutes/NoteRouteTests.cs:
```diff
@@ -0,0 +1,33 @@
+        public void GetNoteForCollaboratorRoute()
+        {
+            url = url.ForGetNoteForCollaborator();
+            var routes = GetRoutes();
```
Can you prevent code duplication?

Rudeka replied:
I managed to work correctly with routes from attributes only by calling GetRoutes method each time when url for a specific route is set and now I have to call it for each test.

Rudeka committed:
Change Identity extension, name for data

TerraVenil commented on DevelopGitHub/GitHubExtension/GitHubExtension.Notes.Tests/Extensions/IdentityExtensions.cs:
```diff
@@ -0,0 +1,17 @@
+{
+    public static class IdentityExtensions
+    {
+        public static void SetUserForController(this ApiController controller, string name, string typeOfClaim, string valueOfClaim)
```
Please prevent code duplication code after rebase.

Rudeka replied:
I have renamed the class to TestsIdentityExtensions and commited

TerraVenil commented on DevelopGitHub/GitHubExtension/GitHubExtension.Notes.Tests/TestsForValidator/NoteValidatorTests.cs:
```diff
@@ -0,0 +1,37 @@
+    public class NoteValidatorTests
+    {
+        [Fact]
+        public void ShoulHaveErrorWhenBodyIsNull()
```
Where the positive test?

Rudeka replied:
There are positive tests below e.g. ShouldNotHaveErrorWhenBodyIsSpecified. I think it is better to rename it something like ValidModelWhenBodyIsSpecified. Made changes and commited.

TerraVenil replied:
Good idea!

Rudeka committed:
Renamed tests for validators

	modified:   DevelopGitHub/GitHubExtension/GitHubExtension.Notes.Tests/TestsForValidator/NoteValidatorTests.cs

Rudeka committed:
Changes after rebase with stylecop
